### ISSUE TITLE
listen for new requests to set connection: close when closing

### ISSUE THREAD

**Issue by dcolens** [tellnes/http-close#2]
If I read the code right, when close is called, it loops into the connections and sets the "connection: close" header for ongoing requests, but what about new requests coming in? It would be nice that by default all incoming request have the header added once close() it called.

**Comment by tellnes** [comment]
It also calls the original `server.close()` method which stops the server from accepting new connections. So there should not be any new requests after you call it.

https://nodejs.org/api/http.html#http_server_close_callback

**Comment by dcolens** [comment]
Yes, for closed connection that is true, but for idle keepalive connections, server-close does nothing. And those might still receive incoming request after server.close() is called.

**Comment by tellnes** [comment]
The library also closes all idle keepalive connections when called (linje 37).

**Comment by tellnes** [comment]
But yes, connections where headers are sent but the request is still sending data (streaming) might actually create new requests. It should be possible to add some code that adds the header to future requests on those connections.

**Comment by dcolens** [comment]
yes, that's what I'm after :)

**Comment by tellnes** [comment]
If someone writes a test for this I might look into implementing this.

**Comment by gajus** [comment]
This handled in https://github.com/gajus/http-terminator/commit/7ced4f726909765fd74b0b92f5ddb3e1551258ab
